src/daemon/daemon.cpp

```diff
@@ -451,8 +451,28 @@ std::vector<mp::NetworkInterface> validate_extra_interfaces(const mp::LaunchRequ
     return interfaces;
 }
 
-auto validate_create_arguments(const mp::LaunchRequest* request, const mp::VirtualMachineFactory& factory)
+void validate_image(const mp::LaunchRequest* request, const mp::VMImageVault& vault,
+                    mp::VMWorkflowProvider& workflow_provider)
```

Comment:
Could be `const`, right?

Reply:
No, I tried that first, but we call `info_for` which isn't `const` (it changes `needs_update`).

Reply:
Ah, right, ok.